config/src/keys.rs

```diff
@@ -117,3 +116,24 @@ where
         self.private_key.take()
     }
 }
+
+pub fn serialize_key<S, K>(key: &K, serializer: S) -> Result<S::Ok, S::Error>
+where
+    S: Serializer,
+    K: Serialize + ValidKeyStringExt,
+{
+    key.to_encoded_string()
+        .map_err(<S::Error as serde::ser::Error>::custom)
+        .and_then(|str| serializer.serialize_str(&str[..]))
+}
+
+pub fn deserialize_key<'de, D, K>(deserializer: D) -> Result<K, D::Error>
+where
+    D: Deserializer<'de>,
+    K: ValidKeyStringExt + DeserializeOwned + 'static,
+{
+    let encoded_key: String = Deserialize::deserialize(deserializer)?;
```

Comment:
This could probably be &str instead of a String

Reply:
Resolved in 2nd to last commit.